azurerm/internal/services/eventgrid/event_subscription.go

```diff
@@ -1153,3 +1186,15 @@ func flattenValues(inputKey *string, inputValues *[]interface{}) map[string]inte
 		"values": values,
 	}
 }
+
+func flattenIdentity(input *eventgrid.EventSubscriptionIdentity) []interface{} {
+	if input == nil {
```

Comment:
the API supports a value of `None` for `Type` but it's not defined in the SDK - so we'll need to make this:

```suggestion
	if input == nil || string(input.Type) == "None" {
```